transport/transport-runtime/src/androidTest/java/com/google/android/datatransport/runtime/SchedulerIntegrationTest.java

```diff
@@ -0,0 +1,185 @@
+import org.junit.runner.RunWith;
+import org.mockito.stubbing.Answer;
+
+@RunWith(AndroidJUnit4.class)
+public class SchedulerIntegrationTest {
+  private static final String TEST_KEY = "test";
+  private static final String TEST_VALUE = "test-value";
+  private static final String testTransport = "testTransport";
+  private TransportInternal transportInternalMock = mock(TransportInternal.class);
+  private TransportBackend mockBackend = mock(TransportBackend.class);
+  private TransportBackend mockBackend2 = mock(TransportBackend.class);
+  private BackendRegistry mockRegistry = mock(BackendRegistry.class);
+  private final Context context = InstrumentationRegistry.getInstrumentation().getContext();
+  private final Uploader mockUploader = mock(Uploader.class);
+  private TransportRuntime runtime;
+
+  @Before
+  public void setUp() {
+    TransportRuntime.setInstance(
+        DaggerTestRuntimeComponent.builder()
+            .setApplicationContext(context)
+            .setUploader(mockUploader)
+            .setBackendRegistry(mockRegistry)
+            .setSchedulerConfig(new SchedulerConfig(500, 100000, 500))
+            .setEventClock(() -> 3)
+            .setUptimeClock(() -> 1)
+            .build());
+    when(mockBackend.decorate(any()))
+        .thenAnswer(
+            (Answer<EventInternal>)
+                invocation ->
+                    invocation
+                        .<EventInternal>getArgument(0)
+                        .toBuilder()
+                        .addMetadata(TEST_KEY, TEST_VALUE)
+                        .build());
+    when(mockBackend2.decorate(any()))
+        .thenAnswer(
+            (Answer<EventInternal>)
+                invocation ->
+                    invocation
+                        .<EventInternal>getArgument(0)
+                        .toBuilder()
+                        .addMetadata(TEST_KEY, TEST_VALUE)
+                        .build());
+    runtime = TransportRuntime.getInstance();
+  }
+
+  String getBackendName() {
+    byte[] array = new byte[8]; // length is bounded by 8
+    new Random().nextBytes(array);
+    return new String(array, Charset.forName("UTF-8"));
+  }
+
+  @Test
+  public void firstAttemptSchedule() {
+
+    String mockBackendName = getBackendName();
+    when(mockRegistry.get(mockBackendName)).thenReturn(mockBackend);
+    TransportFactory factory = runtime.newFactory(mockBackendName);
+    Transport<String> transport =
+        factory.getTransport(testTransport, String.class, String::getBytes);
+    Event<String> stringEvent = Event.ofTelemetry("TelemetryData");
+    EventInternal expectedEvent =
+        EventInternal.builder()
+            .setEventMillis(3)
+            .setUptimeMillis(1)
+            .setTransportName(testTransport)
+            .setPayload("TelemetryData".getBytes())
+            .build();
+    transport.send(stringEvent);
+    verify(mockBackend, times(1)).decorate(eq(expectedEvent));
+    try {
+      TimeUnit.SECONDS.sleep(5);
```

Comment:
Try using `SystemClock.sleep()` instead, that way you won't have to deal with interrupts.